ckanext/pages/actions.py

```diff
@@ -164,14 +129,15 @@ def _pages_update(context, data_dict):
         setattr(out, item, data.get(item,'page' if item =='page_type' else None)) #backward compatible with older version where page_type does not exist
 
     extras = {}
+
     extra_keys = set(schema.keys()) - set(items + ['id', 'created'])
     for key in extra_keys:
         if key in data:
             extras[key] = data.get(key)
     out.extras = json.dumps(extras)
 
     out.modified = datetime.datetime.utcnow()
-    out.user_id = p.toolkit.c.userobj.id
+    out.user_id = context['user']
```

Comment:
I know actions using `c` is annoying and makes the tests fail, but this is cheating :) 
If we say we store an id, we should store an id and not a name, even if it won't break anything. I refactored it a bit on 03221ea